Source/DafnyCore/AST/Statements/ControlFlow/IfStmt.cs

```diff
@@ -27,7 +27,7 @@ public IfStmt(Cloner cloner, IfStmt original) : base(cloner, original) {
     Els = cloner.CloneStmt(original.Els, false);
   }
 
-  public IfStmt(IOrigin origin, bool isBindingGuard, Expression guard, BlockStmt thn, Statement els)
+  public IfStmt(IOrigin origin, bool isBindingGuard, Expression? guard, BlockStmt thn, Statement els)
```

Comment:
`guard` should be marked as nullable in the syntax constructor too. Also, `els` in this constructor should be marked as nullable.